controllers/storagecluster/reconcile.go

```diff
@@ -371,6 +371,17 @@ func (r *StorageClusterReconciler) reconcilePhases(
 		// The object is marked for deletion
 		instance.Status.Phase = statusutil.PhaseDeleting
 
+		storageClusterPeerList := ocsv1.StorageClusterPeerList{}
+		err := r.List(ctx, &storageClusterPeerList, client.InNamespace(instance.Namespace))
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if len(storageClusterPeerList.Items) != 0 {
+			err := fmt.Errorf("waiting for %v StorageClusterPeer attached to StorageCluster to be deleted before proceeding", len(storageClusterPeerList.Items))
+			r.Log.Info(err.Error())
```

Comment:
error logging is error handling and don't do it twice, I suppose this log line will appear twice, once in this Info and another as error in reconciler.